Description: Fixup handling of init detection for newer Ubuntu releases
Author: James Page <james.page@ubuntu.com>
Forwarded: https://github.com/ceph/ceph/pull/6873

--- a/src/ceph-detect-init/ceph_detect_init/debian/__init__.py
+++ b/src/ceph-detect-init/ceph_detect_init/debian/__init__.py
@@ -9,5 +9,8 @@ def choose_init():
     Returns the name of a init system (upstart, sysvinit ...).
     """
     if distro.lower() in ('ubuntu', 'linuxmint'):
-        return 'upstart'
+        if codename >= 'vivid':
+            return 'systemd'
+        else:
+            return 'upstart'
     return 'sysvinit'
--- a/src/ceph-detect-init/tests/test_all.py
+++ b/src/ceph-detect-init/tests/test_all.py
@@ -50,6 +50,9 @@ class TestCephDetectInit(testtools.TestC
         with mock.patch('ceph_detect_init.debian.distro',
                         'ubuntu'):
             self.assertEqual('upstart', debian.choose_init())
+            with mock.patch('ceph_detect_init.debian.codename',
+                            'vivid'):
+                self.assertEqual('systemd', debian.choose_init())
 
     def test_fedora(self):
         with mock.patch('ceph_detect_init.fedora.release',
